src/main/scala/package.scala

```diff
@@ -1,21 +1,24 @@
 package com.originate.scalypher
 
+import com.originate.scalypher.action.ActionNodeOrRelationship
 import com.originate.scalypher.action.ActionReference
 import com.originate.scalypher.where.ObjectReference
-import com.originate.scalypher.where.Reference
+import com.originate.scalypher.where.NodeOrRelationshipReference
 
 package object types {
-  trait Referenceable {
-    def as(name: String): ActionReference =
-      ActionReference(ObjectReference(this), Some(name))
+  trait Referenceable
```

Comment:
right, what i was thinking is we have something vaguely like...
```
sealed trait Identifiable
trait Referenceable
trait Unreferenceable {
  ...
}

// and in some other file.scala
case class Path extends Unreferenceable {
  ...
}
```

some quick testing just now suggests that would work, and gives us that guarantee over `Identifiable` at least